evaluation/evaluate.py

```diff
@@ -259,52 +288,70 @@ def plot_diagrams(questions_with_ratings: list, requested_metrics: list, passing
         "mean_rating": "Rating Score",
     }
 
-    stats_y_lim = {"pass_count": len(questions_with_ratings), "pass_rate": 1.0, "mean_rating": 5.0}
-
+    stats_y_lim = {
+        "pass_count": len(questions_with_ratings_dict[next(iter(questions_with_ratings_dict))]),
+        "pass_rate": 1.0,
+        "mean_rating": 5.0,
+    }
     # Draw the chart for the results
-    data = [data for _, data in rating_stat_data.items()]
-    titles = [display_stats_name[mn] for mn in rating_stat_data.keys()]
-    y_labels = [stats_y_labels[mn] for mn in rating_stat_data.keys()]
-    y_lims = [stats_y_lim[mn] for mn in rating_stat_data.keys()]
-    layout = (int(np.ceil(len(rating_stat_data) / 3)), 3 if len(rating_stat_data) > 3 else len(rating_stat_data))
+    data = {}
+    for key in questions_with_ratings_dict:
+        data_per_model = [data for _, data in rating_stat_data[key].items()]
+        data[key] = data_per_model
+        titles = [display_stats_name[mn] for mn in rating_stat_data[key].keys()]
+        y_labels = [stats_y_labels[mn] for mn in rating_stat_data[key].keys()]
+        y_lims = [stats_y_lim[mn] for mn in rating_stat_data[key].keys()]
+        layout = (
+            int(np.ceil(len(rating_stat_data[key]) / 3)),
+            3 if len(rating_stat_data[key]) > 3 else len(rating_stat_data[key]),
+        )
 
     plot_bar_charts(
-        layout, data, titles, y_labels, results_dir / "evaluation_results.pdf", y_max_lim=y_lims, width=width
+        layout,
+        data,
+        titles,
+        y_labels,
+        results_dir / "evaluation_results.png",
+        y_lims,
+        width,
     )
 
-    gpt_metric_avg_ratings = [val for _, val in rating_stat_data["mean_rating"].items()]
+    gpt_metric_avg_ratings = {}
+    data_for_single_box = {}
+    data_for_multi_box = {}
+    for key in questions_with_ratings_dict:
+        gpt_metric_avg_ratings[key] = [val for _, val in rating_stat_data[key]["mean_rating"].items()]
+        data_for_single_box[key] = [data for _, data in gpt_metric_data_points[key].items()]
+        data_for_multi_box[key] = [data for _, data in stat_metric_data_points[key].items()]
```

Comment:
If you only need the values (not the keys) you can retrieve them like this:
```suggestion
        gpt_metric_avg_ratings[key] = [val for val in rating_stat_data[key]["mean_rating"].values()]
        data_for_single_box[key] = [data for data in gpt_metric_data_points[key].values()]
        data_for_multi_box[key] = [data for data in stat_metric_data_points[key].values()]
```

But then, do you even need the list comprehension? If you can't use the object returned by `values()` directly, you can still do sth like this:

```suggestion
        gpt_metric_avg_ratings[key] = list(rating_stat_data[key]["mean_rating"].values())
        data_for_single_box[key] = list(gpt_metric_data_points[key].values())
        data_for_multi_box[key] = list(stat_metric_data_points[key].values())
```